src/scripts/modules/destinations.js

```diff
@@ -0,0 +1,136 @@
+import destinationsList from "../destinationList.js";
+import events from "./pubsub.js";
+
+// Create  and display destination lists
+
+// Set the "shared" variables among different methods:
+let buttons;
+let visited;
+let bucketlist;
+let all;
+let gallery;
+let activeBtn;
+
+function cacheDom() {
+  buttons = Array.from(document.querySelector(".buttons").children);
+  visited = buttons[1];
+  bucketlist = buttons[2];
+  all = buttons[3];
+  gallery = document.querySelector(".gallery");
+  activeBtn = null;
+}
+
+function bindEvents() {
+  visited.addEventListener("click", addVisited);
+  bucketlist.addEventListener("click", addBucketlist);
+  all.addEventListener("click", showAll);
+  document.addEventListener("DOMContentLoaded", setActive);
+  document.addEventListener("scroll", shrinkNav);
+  window.addEventListener("scroll", restoreNav);
+  events.subscribe("destinationAdded", renderDestination);
```

Comment:
This is 99% done! Did you realize what was missing? Doing a console.log inside the callback would help you to realize what was missing and if this was being called or not.

The way it was created, `renderDestination` expects the destination data as a parameter, so that it knows what to build. That part is missing here. Then,  it returns an element, and we need to append it to the DOM. Similar to how you've done it in `render()` method. Without those 2 things, this wouldn't work.

```js
  events.subscribe("destination_added", (index) => {
    // Render the destination item based on the index passed
    const elDestination = renderDestination(destinationsList[index]);
    gallery.appendChild(elDestination);
  });
```

And voilá!

Reply:
Yep I knew what was missing but not how to make it work. The tutorial I followed was super straightforward and made me believe I should could for something as simple

```
export const movies = {
  list: [],
  render: container => {
    let template = document.getElementById('movieListTemplate');
    ...............................................
    //listen for movieAdded messages
    pubsub.subscribe('movieAdded', movies.movieAdded);
  },
```
Basically my knowledge is still pretty limited because when I saw 'function' in 'subscribe' it never crossed my mind it can be done the way you did it here

Reply:
This should now work if I'm not wrong but main.js is giving me this error ```Uncaught TypeError: can't access property "visited", destination is undefined``` did I not import something correctly...?

Reply:
Yes, you could also do it that way. In our case:
```js
    pubsub.subscribe('destinationAdded', formState);
```

But since we already have access to in the `destinations` module, you can just pass the index :)


-------

> Uncaught TypeError: can't access property "visited", destination is undefined

It doesn't seem as a import problem. Is the `destination` variable defined in the scope? 
Can you show a screenshot with the error, which line does that happen? In what exact moment? If you push the code it's easier for me to see it. :)